File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
import java.util.concurrent.ConcurrentHashMap;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        ConcurrentHashMap<String,Set<String>> set = new ConcurrentHashMap<>();
        for (String[] c : clothes) {
            set.computeIfAbsent(c[1], k -> new HashSet<>()).add(c[0]);
        }
        
       
        for (Set<String> clothesSet : set.values()) {
            answer *= (clothesSet.size() + 1);
        }
    
        
        
        return answer -1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    hashclothes = {}
    for c in clothes:
        if c[1] not in hashclothes:
            hashclothes[c[1]] = set()
        hashclothes[c[1]].add(c[0])
        
    for clothe in hashclothes.values():
        answer *= len(clothe)+1
    return answer-1